OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -118,20 +119,24 @@ public Harvester(Actor self, HarvesterInfo info)
 
 		void INotifyCreated.Created(Actor self)
 		{
+			notify = self.TraitsImplementing<INotifyHarvesterAction>().ToArray();
+
 			if (Info.SearchOnCreation)
 				self.QueueActivity(new FindResources(self));
 		}
 
 		void INotifyBuildComplete.BuildingComplete(Actor self)
 		{
+			notify = self.TraitsImplementing<INotifyHarvesterAction>().ToArray();
```

Comment:
Ok, in that case I have two suggestions:
* Try wrapping the QueueActivity from Created in a FrameEndTask to see if that makes it run late enough to not be overridden.
* If that fails, implement INotifyAddedToWorld, with a flag so that the code only runs the first time it is added.

Implementing BuildingComplete as a bandaid for the code run from Created not working all the time is ugly, and is something that we really should fix.